pkg/kv/kvserver/gc/gc_random_test.go

```diff
@@ -117,6 +117,9 @@ func TestRunNewVsOld(t *testing.T) {
 		},
 	} {
 		t.Run(fmt.Sprintf("%v@%v,ttlSec=%v", tc.ds, tc.now, tc.ttlSec), func(t *testing.T) {
+			rng, seed := randutil.NewTestRand()
+			t.Logf("running with random seed %d", seed)
```

Comment:
I think the usual way to do this is to call `randutil.SeedForTests()` in `TestMain` in each package, which will output the seed for the package. That will also seed the global `rand` RNG.

Reply:
Looked on implementation of SeedForTests() and #71350. There's a slight issue with this as it doesn't work very well with nested tests. And we use nested tests everywhere in kv/storage.

In brief you have to rerun all subtests sequentially to get the same seed as you saw in the logs when TestMain runs. That's because the seed from ENV is used to run global rng and its results are used to seed individual tests. With nested tests it makes it painful to debug as you can't rerun individual tests with conditional breakpoints to avoid stepping through all invocations.

Reply:
I decided to keep logging, but added proper initialization on the package level. So batch reruns should work from bazel and you can just hack test code to rerun subtest with appropriate seed as needed.

Reply:
Ah, too bad, would've been nice if this was just handled somehow. Anyway, this seems fine, thanks for looking into it.